relay-server/src/metrics_extraction/transactions.rs

```
@@ -169,6 +169,35 @@ fn extract_user_satisfaction(
     None
 }
 
+/// Decide whether we want to keep the transaction name.
+/// High-cardinality sources are excluded to protect our metrics infrastructure.
+/// Note that this will produce a discrepancy between metrics and raw transaction data.
+#[cfg(feature = "processing")]
+fn keep_transaction_name(source: &TransactionSource) -> bool {
+    match source {
+        // "custom" is like a box of chocolates, you never know what you're gonna get.
+        // For now, we hope that custom transaction names set by users are low-cardinality.
+        TransactionSource::Custom => true,
+        // "url" are raw URLs, potentially containing identifiers.
+        TransactionSource::Url => false,
+        TransactionSource::Route => true,
+        TransactionSource::View => true,
+        TransactionSource::Component => true,
+        TransactionSource::Task => true,
+
+        // "unknown" is the value for old SDKs that do not send a transaction source yet.
+        // Assume high-cardinality and drop.
+        TransactionSource::Unknown => false,
+
+        // Any other value would be an SDK bug, assume high-cardinality and drop.
+        TransactionSource::Other(source) => {
+            // EnvelopeProcessor sets project and SDK on the scope, so no need to add sentry tags here.
+            relay_log::error!("Invalid transaction source: '{}'", source);
```

Not sure if it really makes sense to log an error here. If an SDK adds a transaction source deliberately, we will have to ignore this error until Relay updates.